src/groupeddataframe/show.jl

```diff
@@ -35,36 +35,66 @@ function Base.show(io::IO, gd::GroupedDataFrame;
                  truncate=truncate, kwargs...)
         end
     else
-        if N > 0
-            nrows = size(gd[1], 1)
-            rows = nrows > 1 ? "rows" : "row"
+        N > 0 || return
 
-            identified_groups = [string(col, " = ", repr(gd[1][1, col]))
-                                 for col in gd.cols]
 
-            print(io, "\nFirst Group ($nrows $rows): ")
-            join(io, identified_groups, ", ")
-            println(io)
+        (h, w) = get(io, :displaysize, displaysize(io))
```

Comment:
@ronisbr - do you use the same pattern in PrettyTables.jl? (to ensure consistency)

@Jollywatt - should not we use the `get(io, :limit, false)` test first and only implement the logic you propose if the `:limit` property is set? And add tests for both cases.

Note the following for standard printing of data frames:
```
julia> df = DataFrame(rand(20, 5), :auto) # display sets limit
20×5 DataFrame
 Row │ x1        x2         x3          x4         x5
     │ Float64   Float64    Float64     Float64    Float64
─────┼───────────────────────────────────────────────────────
   1 │ 0.799571  0.711126   0.0829037   0.153365   0.65724
   2 │ 0.013505  0.516158   0.289836    0.772033   0.236464
   3 │ 0.539347  0.866731   0.527391    0.763274   0.445696
  ⋮  │    ⋮          ⋮          ⋮           ⋮          ⋮
  18 │ 0.285917  0.83837    0.412981    0.283734   0.734432
  19 │ 0.539428  0.281714   0.503087    0.236329   0.980165
  20 │ 0.19463   0.818894   0.080396    0.274849   0.678159
                                              14 rows omitted

julia> show(stdout, df) # stdout does not have limit
20×5 DataFrame
 Row │ x1        x2         x3          x4         x5
     │ Float64   Float64    Float64     Float64    Float64
─────┼────────────────────────────────────────────────────────
   1 │ 0.799571  0.711126   0.0829037   0.153365   0.65724
   2 │ 0.013505  0.516158   0.289836    0.772033   0.236464
   3 │ 0.539347  0.866731   0.527391    0.763274   0.445696
   4 │ 0.220234  0.94367    0.740741    0.141462   0.121854
   5 │ 0.32305   0.267344   0.519681    0.331716   0.396574
   6 │ 0.127974  0.282572   0.640234    0.841006   0.526737
   7 │ 0.322138  0.925825   0.00683361  0.0276174  0.140838
   8 │ 0.832967  0.744335   0.808838    0.516439   0.219536
   9 │ 0.895046  0.853179   0.487567    0.81502    0.721893
  10 │ 0.236999  0.481331   0.310607    0.0200533  0.00837921
  11 │ 0.142544  0.310369   0.121286    0.781769   0.0948759
  12 │ 0.923275  0.205102   0.097269    0.806569   0.273017
  13 │ 0.177222  0.266212   0.965611    0.345097   0.745195
  14 │ 0.157075  0.851299   0.121473    0.281857   0.1338
  15 │ 0.938703  0.183045   0.769088    0.562905   0.0955595
  16 │ 0.592258  0.387368   0.912301    0.292013   0.401606
  17 │ 0.949321  0.0872963  0.421829    0.0586528  0.39613
  18 │ 0.285917  0.83837    0.412981    0.283734   0.734432
  19 │ 0.539428  0.281714   0.503087    0.236329   0.980165
  20 │ 0.19463   0.818894   0.080396    0.274849   0.678159
julia> show(IOContext(stdout, :limit => true), df) # but you can manually enable limit
20×5 DataFrame
 Row │ x1        x2         x3          x4         x5
     │ Float64   Float64    Float64     Float64    Float64
─────┼───────────────────────────────────────────────────────
   1 │ 0.799571  0.711126   0.0829037   0.153365   0.65724
   2 │ 0.013505  0.516158   0.289836    0.772033   0.236464
   3 │ 0.539347  0.866731   0.527391    0.763274   0.445696
  ⋮  │    ⋮          ⋮          ⋮           ⋮          ⋮
  18 │ 0.285917  0.83837    0.412981    0.283734   0.734432
  19 │ 0.539428  0.281714   0.503087    0.236329   0.980165
  20 │ 0.19463   0.818894   0.080396    0.274849   0.678159
                                              14 rows omitted
```

Reply:
> @ronisbr - do you use the same pattern in PrettyTables.jl? (to ensure consistency) 

Yes, kind of. The idea is the same, but I check for `:limit` if the user did not pass a specific keyword to crop the output table.

Reply:
I think it does not have to, see e.g. https://github.com/JuliaLang/julia/blob/d5cde865f24d2c3e7041aee8ea464eb6b6045a2a/base/stream.jl#L564 (and custom targets could define similar rules).

OTOH, I do not see a situation when what is proposed now would be incorrect.

Reply:
Neither DataFrames nor PrettyTables use that pattern AFAICT. Julia Base doesn't. So better be consistent.

Reply:
DataFrames.jl uses this pattern in https://github.com/JuliaData/DataFrames.jl/blob/8acb679cafc44c2d6b524fbaf4df7809f9ded8f2/src/abstractdataframe/io.jl#L389

Blame shows that I added it 3 years ago in https://github.com/JuliaData/DataFrames.jl/pull/1761. The reason why it was added is https://github.com/JuliaData/DataFrames.jl/pull/1761#discussion_r275116264 (you had the same comment there as here). @nalimilan - so what do we do?